### PR TITLE
[App] Actually fix launching apps on multiple clusters

### PR DESCRIPTION
**Pull request by luca3rd**
## What does this PR do?

Actually fix the cross launching apps issue.

### After
Screen Shot 2022-11-01 at 9 25 34 PM

Fixes #\<issue_number>

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [ ] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by lantiga** [comment]
I don't think this is what we want.

App names must be unique per-project, not per-cluster @edenafek

**Comment by rusenask** [comment]
> I don't think this is what we want.
> 
> App names must be unique per-project, not per-project + per-cluster @edenafek (I feel strongly about this).
> 
> If you try to run an app with the same name in a project, targeting a different cluster, the first app needs to be stopped, irrespective of what cluster it is running on. We need to keep the model simple, otherwise the CLI will complicate a ton (now it's not enough to address the app by name, you also need to think whether there's another instance of that app running on another cluster, and instead use the id to address it, e.g. connect, from the CLI).

:+1: I also don't think this is useful for any scenario. Why would you have the same name @edenafek ? If I wanted this as a user I would just create two apps `muse-east-1` and `muse-east-2` and then I don't have to think about it. This is almost as having two files in your directory with the same name, while possible, doesn't make things good :)

File: src/lightning_app/runners/cloud.py [hunk 1]
```diff
@@ -285,14 +285,25 @@ def dispatch(
...
+            if previous_app_instances:
+                most_recent_instance = previous_app_instances[0]
                 if not app_config.cluster_id:
-                    # Re-run the app on the same cluster
-                    app_config.cluster_id = existing_instance.spec.cluster_id
+                    # Re-run the app on the most recent cluster
+                    app_config.cluster_id = most_recent_instance.spec.cluster_id
+
+                # get the existing instance on the cluster, if any
+                existing_instance = existing_instances_by_cluster.get(app_config.cluster_id, None)
```
Comment (dmitsf) [inline src/lightning_app/runners/cloud.py:303]:
```suggestion
                existing_instance = existing_instances_by_cluster.get(app_config.cluster_id)
```

Comment (dmitsf) [reply src/lightning_app/runners/cloud.py:303]: `None` is already the default returning value :)

**Pull request closed without merging**